core/src/apps/cardano/layout.py

```diff
@@ -401,31 +401,24 @@ async def show_warning_address_foreign_staking_key(
     staking_account_path: List[int],
     staking_key_hash: bytes,
 ) -> None:
-    await show_warning(
-        ctx,
-        (
-            "Stake rights associated",
-            "with this address do",
-            "not match your",
-            "account",
-            address_n_to_str(account_path),
-        ),
-        button="Ok",
-    )
-
+    page1 = Text("Warning", ui.ICON_WRONG, ui.RED)
+    page1.normal("Stake rights associated")
+    page1.normal("with this address do")
+    page1.normal("not match your")
+    page1.normal("account %s:" % format_account_number(account_path))
```

Comment:
this still seems untrue when the path is non-standard, but that's presumably OK with the full derivation path shown anyway